src/wasm.rs

```diff
@@ -223,3 +223,13 @@ pub fn platform() -> Platform {
         Platform::Unknown(string.to_string())
     }
 }
+
+pub fn arch() -> Arch {
+    if cfg!(target_pointer_width = "32") {
+        Arch::Wasm32
+    } else if cfg!(target_pointer_width = "64"){
+        Arch::Wasm64
+    } else {
+        unreachable!("Arches other than wasm32 and wasm64 are not supported")
+    }
```

Comment:
Yep, consistency makes things better

Reply:
Replaces `unreachable!()` with `compile_error!()`?

```rust
pub fn arch() -> Arch {
    if cfg!(target_pointer_width = "32") {
        Arch::Wasm32
    } else if cfg!(target_pointer_width = "64"){
        Arch::Wasm64
    } else {
        compile_error!("Arches other than wasm32 and wasm64 are not supported")
    }
}
```

With this change, I can not cross-compile this crate to `asmjs-unknown-emscripte` and `wasm32-unknown-emscripten`:

```shell
$ cargo b --target wasm32-unknown-emscripten
   Compiling whoami v1.2.3 (/home/steve/Documents/workspace/whoami)
error: Arches other than wasm32 and wasm64 are not supported
   --> src/wasm.rs:233:9
    |
233 |         compile_error!("Arches other than wasm32 and wasm64 are not supported")
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `whoami` due to previous error

$ cargo b --target asmjs-unknown-emscripten 
error: Arches other than wasm32 and wasm64 are not supported
   --> src/wasm.rs:233:9
    |
233 |         compile_error!("Arches other than wasm32 and wasm64 are not supported")
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `whoami` due to previous error

Reply:
I think the problem is that you need to use `#[cfg()]` instead of `cfg!()`

```rust
pub fn arch() -> Arch {
    #[cfg(target_pointer_width = "32")] {
        Arch::Wasm32
    }

    #[cfg(target_pointer_width = "64")] {
        Arch::Wasm64
    }

    #[cfg(not(any(target_pointer_width = "64", target_pointer_width = "32")))] {
        compile_error!("Arches other than wasm32 and wasm64 are not supported")
    }
}
```